gm4_lively_lily_pads/data/gm4_lively_lily_pads/function/mechanics/interactions/placement/interact_rcd.mcfunction

```diff
@@ -0,0 +1,44 @@
+# place interaction logic
+# @s = player who right clicked on interaction holding a supported item
+# at @s
+# run from advancement: placement_rcd
+
+advancement revoke @s only gm4_lively_lily_pads:placement_rcd
+
+# breif check to enable Upgrade_Path 3.0
+execute if entity @e[type=minecraft:interaction,tag=lilyPadInt,limit=1] run scoreboard players set lively_lily_pads gm4_earliest_version 0
+
+# gamemode flags
+execute if entity @s[gamemode=adventure] run scoreboard players set $adventure gm4_llp.data 1
+
+# item information
+execute if items entity @s weapon.mainhand #gm4_lively_lily_pads:coral_fan run \
+  scoreboard players set $item_type gm4_llp.data 1
+execute if items entity @s weapon.mainhand #gm4_lively_lily_pads:dead_coral_fan run \
+  scoreboard players set $item_type gm4_llp.data 2
+execute if items entity @s weapon.mainhand minecraft:torch run \
+  scoreboard players set $item_type gm4_llp.data 3
+execute if items entity @s weapon.mainhand minecraft:lantern run \
+  scoreboard players set $item_type gm4_llp.data 4
+execute if items entity @s weapon.mainhand minecraft:soul_torch run \
+  scoreboard players set $item_type gm4_llp.data 5
+execute if items entity @s weapon.mainhand minecraft:soul_lantern run \
+  scoreboard players set $item_type gm4_llp.data 6
+execute if items entity @s weapon.mainhand #minecraft:candles run \
+  scoreboard players set $item_type gm4_llp.data 7
+execute if items entity @s weapon.mainhand minecraft:cactus_flower run \
+  scoreboard players set $item_type gm4_llp.data 8
+execute if items entity @s weapon.mainhand minecraft:redstone_torch run \
+  scoreboard players set $item_type gm4_llp.data 9
+execute if items entity @s weapon.mainhand minecraft:spore_blossom run \
+  scoreboard players set $item_type gm4_llp.data 10
+
```

Comment:
There's probably a nicer way of doing this, tho I can't think of it right now either.